File: android/src/ti/airship/utils/PluginLogger.java
/* Copyright Urban Airship and Contributors */

package ti.airship.utils;

import android.util.Log;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.annotation.RestrictTo;

import com.urbanairship.LoggingCore;

/**
 * Cordova logger for Urban Airship.
 */
public final class PluginLogger {

    private static LoggingCore logger = new LoggingCore(Log.ERROR, "UALib-Ti.Airship");

    /**
     * Private, unused constructor
     */
    private PluginLogger() {
    }

    public static void logCall(@NonNull String name, @Nullable Object... args) {
        debug("%s: %s", name, args);
    }

    public static void setLogLevel(int logLevel) {
        logger.setLogLevel(logLevel);
    }

    public static void warn(@NonNull String message, @Nullable Object... args) {
        logger.log(Log.WARN, null, message, args);
    }

    public static void warn(@NonNull Throwable t, @NonNull String message, @Nullable Object... args) {
        logger.log(Log.WARN, t, message, args);
    }

    public static void warn(@NonNull Throwable t) {
        logger.log(Log.WARN, t, null);
    }

    public static void verbose(@NonNull String message, @Nullable Object... args) {
        logger.log(Log.VERBOSE, null, message, args);
    }

    public static void debug(@NonNull String message, @Nullable Object... args) {
        logger.log(Log.DEBUG, null, message, args);
    }

    public static void debug(@NonNull Throwable t, @NonNull String message, @Nullable Object... args) {
        logger.log(Log.DEBUG, t, message, args);
    }

    public static void info(@NonNull String message, @NonNull Object... args) {
        logger.log(Log.INFO, null, message, args);
    }

    public static void info(@NonNull Throwable t, @NonNull String message, @Nullable Object... args) {
        logger.log(Log.INFO, t, message, args);
    }

    public static void error(@NonNull String message, @Nullable Object... args) {
        logger.log(Log.ERROR, null, message, args);
    }

    public static void error(@NonNull Throwable t) {
        logger.log(Log.ERROR, t, null);
    }

    public static void error(@NonNull Throwable t, @NonNull String message, @Nullable Object... args) {
        logger.log(Log.ERROR, t, message, args);
    }
}

File: android/src/ti/airship/utils/PluginLogger.kt
/* Copyright Urban Airship and Contributors */

package ti.airship.utils

import android.util.Log
import com.urbanairship.LoggingCore

/**
 * Cordova logger for Urban Airship.
 */
object PluginLogger {
    private val logger = LoggingCore(Log.ERROR, "UALib-Ti.Airship")
    fun logCall(name: String, vararg args: Any?) {
        debug("$name: ${args.contentDeepToString()}")
    }

    fun setLogLevel(logLevel: Int) {
        logger.logLevel = logLevel
    }

    fun warn(message: String, vararg args: Any?) {
        logger.log(Log.WARN, null, message, *args)
    }

    fun warn(t: Throwable, message: String, vararg args: Any?) {
        logger.log(Log.WARN, t, message, *args)
    }

    fun warn(t: Throwable) {
        logger.log(Log.WARN, t, null)
    }

    fun verbose(message: String, vararg args: Any?) {
        logger.log(Log.VERBOSE, null, message, *args)
    }

    fun debug(message: String, vararg args: Any?) {
        logger.log(Log.DEBUG, null, message, *args)
    }

    fun debug(t: Throwable, message: String, vararg args: Any?) {
        logger.log(Log.DEBUG, t, message, *args)
    }

    fun info(message: String, vararg args: Any) {
        logger.log(Log.INFO, null, message, *args)
    }

    fun info(t: Throwable, message: String, vararg args: Any?) {
        logger.log(Log.INFO, t, message, *args)
    }

    fun error(message: String, vararg args: Any?) {
        logger.log(Log.ERROR, null, message, *args)
    }

    fun error(t: Throwable) {
        logger.log(Log.ERROR, t, null)
    }

    fun error(t: Throwable, message: String, vararg args: Any?) {
        logger.log(Log.ERROR, t, message, *args)
    }


}